python/dolfinx/fem/function.py

```diff
@@ -83,9 +84,9 @@ def __complex__(self):
 
 
 class Expression:
-    def __init__(self, ufl_expression: ufl.core.expr.Expr, X: np.ndarray,
-                 form_compiler_options: dict = {}, jit_options: dict = {},
-                 dtype=default_scalar_type):
+    def __init__(self, e: ufl.core.expr.Expr, X: np.ndarray,
+                 comm: typing.Optional[_MPI.Comm] = None, form_compiler_options: dict = {},
+                 jit_options: dict = {}, dtype=None):
```

Comment:
We should really use `jit_options: Optional[dict] = None`, ref https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python